From: Jia Chao <jiachao2130@126.com>
Date: Tue, 13 Apr 2021 10:07:54 +0800
Subject: keyboard: Fix suggestions box

The suggestions box actor was not being correctly populated.
Make it correctly use the St.BoxLayout add_child() method.

Fixes: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/3422
Fixes: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/2965
Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1806>
(cherry picked from commit f76e6877c7f788b08a1bf371bcc49207eda4882d)

Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/commit/d3eecd6d
---
 js/ui/keyboard.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/ui/keyboard.js b/js/ui/keyboard.js
index 486232d..e2574f0 100644
--- a/js/ui/keyboard.js
+++ b/js/ui/keyboard.js
@@ -197,7 +197,7 @@ class Suggestions extends St.BoxLayout {
     add(word, callback) {
         let button = new St.Button({ label: word });
         button.connect('clicked', callback);
-        this.add(button);
+        this.add_child(button);
     }
 
     clear() {
